### PR TITLE
Link the VideoVIewModel to the exercise screen

### PR DESCRIPTION
**Pull request by Shayan105**
# Summary 
This PR completes the workout UI of the app by showing the correct video for each exercise during a workout session.

#Modified files:
- Main activity.kt 
- WorkoutScreen.kt
- WorkoutScreenTest.kt

#Added test:
-WorkoutScreenTest.kt : Test that the screen works for every type of workout (Body-Weight, Yoga and Warmup) and check that the VIdeoViewModel is used properly

# Remark:
1)The screen shows a loading circle if the videoList is not fetched. 
2) Videos are fetched on demand. It means that the VideoViewModel fetches a list of URLs that point to the respective videos on Firebase.

### CODE REVIEW

**Review by didi2101 — CHANGES_REQUESTED** [review]
Nice small PR with good changes !

Overall very good, i like the idea of the loading Screen and the URL implementation.

Just very (very) small changes to do

File: app/src/androidTest/java/com/android/sample/ui/workout/WorkoutScreenTest.kt [hunk 1]
```diff
@@ -148,19 +194,124 @@ class WorkoutScreenTest {
...
+    composeTestRule.onNodeWithTag("SkipButton").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("SkipButton").assertTextEquals("Skip")
+
+    composeTestRule.onNodeWithTag("StartButton").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("StartButton").assertTextEquals("Start")
+    composeTestRule.onNodeWithTag("recordSwitch").assertIsDisplayed()
+
+    // VideoPlayer
+    composeTestRule.onNodeWithTag("VideoPlayer").assertIsDisplayed()
+
```
Comment (didi2101) [inline app/src/androidTest/java/com/android/sample/ui/workout/WorkoutScreenTest.kt:243]: Nice test, very clear !

File: app/src/androidTest/java/com/android/sample/ui/workout/WorkoutScreenTest.kt [hunk 2]
```diff
@@ -148,19 +194,124 @@ class WorkoutScreenTest {
...
+    // ClickOnStart to start the 1st activity
+    composeTestRule.onNodeWithTag("StartButton").performClick()
```
Comment (didi2101) [inline app/src/androidTest/java/com/android/sample/ui/workout/WorkoutScreenTest.kt:297]: Why click here in this test ? 

I think you could remove it

File: app/src/androidTest/java/com/android/sample/ui/workout/WorkoutScreenTest.kt [hunk 3]
```diff
@@ -148,19 +194,124 @@ class WorkoutScreenTest {
...
+          bodyweightViewModel = bodyWeightViewModel,
+          yogaViewModel = yogaViewModel,
+          warmUpViewModel = warmUpViewModel,
+          workoutType = WorkoutType.BODY_WEIGHT,
+          videoViewModel = mockVideoViewModel)
+    }
+
+    // ClickOnStart to start the 1st activity
+    composeTestRule.onNodeWithTag("StartButton").performClick()
+    verify(mockVideoRepository, atLeast(1)).getVideos(any(), any())
```
Comment (didi2101) [inline app/src/androidTest/java/com/android/sample/ui/workout/WorkoutScreenTest.kt:314]: Nice test !

But the title doesn't say that you verify this.

**Review by didi2101 — CHANGES_REQUESTED** [review]
Very nice tests, that checked all the needed composables are displayed.

You also modify what i asked to change.

You can modify my little modifications.

File: app/src/androidTest/java/com/android/sample/ui/workout/WorkoutScreenTest.kt (hunk 2 above)
Comment (Shayan105) [reply app/src/androidTest/java/com/android/sample/ui/workout/WorkoutScreenTest.kt:297]: This is to start the 1st exercise of the workout

File: app/src/androidTest/java/com/android/sample/ui/workout/WorkoutScreenTest.kt (hunk 3 above)
Comment (Shayan105) [reply app/src/androidTest/java/com/android/sample/ui/workout/WorkoutScreenTest.kt:314]: This actually starts the 1s exercise of the workout, maybe I didnt understand your comment ?

File: app/src/androidTest/java/com/android/sample/ui/workout/WorkoutScreenTest.kt [hunk 4]
```diff
@@ -275,4 +421,36 @@ class WorkoutScreenTest {
...
+          bodyweightViewModel = bodyWeightViewModel,
+          yogaViewModel = yogaViewModel,
+          warmUpViewModel = warmUpViewModel,
+          workoutType = WorkoutType.BODY_WEIGHT,
+          videoViewModel = mockVideoViewModel)
+    }
+    // Check that on launching the composable, the videoViewModel tries to fetch the videos on
+    // firestore
+    verify(mockVideoRepository, atLeastOnce()).getVideos(any(), any())
+  }
```
Comment (didi2101) [inline app/src/androidTest/java/com/android/sample/ui/workout/WorkoutScreenTest.kt:455]: It's better like this

Comment (didi2101) [inline app/src/androidTest/java/com/android/sample/ui/workout/WorkoutScreenTest.kt:455]: It's better like this

**Review by didi2101 — APPROVED** [review]
The changes i asked for have been made

Approved.

**Pull request merged**